sysrib/sysrib.go

```diff
@@ -48,9 +54,71 @@ type NIRIB struct {
 	IPV4 *generics_tree.TreeV4[*Route]
 }
 
+// GUEPolicy represents the static values in the IP and UDP headers that are
+// to encapsulate the packet.
+//
+// srcPort is a hash.
+// dstIP is the nexthop of the BGP route.
+type GUEPolicy struct {
+	dstPort [2]byte
+	srcIP4  [4]byte
```

Comment:
i think this code would be simpler if you had one slice srcIP since you already is ivV6 bool

Reply:
I wanted to use an array type here to keep the types tight.

Reply:
i'm not sure i understand i understand what advantage this has considering you check the boolean every time these are used. you also wouldn't convert to and from slice/array every time you use these

Reply:
Oh I remember now, this type has to be hashable in order to do the diffing of the resolved nexthops, so a slice, or net.IP, wouldn't work.